### PR TITLE
HACKING: avoid infix optic functions, require explicit import lists

### PR DESCRIPTION
**Pull request by frasertweedale**
add some stuff to HACKING.

The import list thing is hopefully uncontroversial.

The optic thing is a strong preference I have (mainly because I can never remember
what they are, but the "word" ones are much more human friendly and therefore readable (IMO).

If you agree, no need to wholesale refactor.  Drive-bys are fine.

### CODE REVIEW

**Comment by romanofski** [comment]
All fine by me :-)

**Comment by frasertweedale** [comment]
@romanofski do you reckon we could rebase PRs instead of merge commits (in future)?
Keeps the history a bit cleaner and easier to follow IMO.

**Comment by romanofski** [comment]
Sounds good. Since we're on the topic: I've been using [stylish-haskell](https://github.com/jaspervdj/stylish-haskell) for most of the imports and function formatting. I think the only thing which irks me is the fact that some stylish changes reformat the entire import block depending on the width of the imports. So I'm not sure if it will make hard to follow changes in the future if they're just clobbered by various "styling" changes. What are your thoughts on this?

**Comment by frasertweedale** [comment]
@romanofski I am against aligning import lists.  If a new (or removed) import
changes the width, everything is then misaligned.

i.e. **do not want** the following:

```
import qualified Foo     (a, b)
import           Bar.Baz (c, d)
```

, because when we need to add `import Quuz.Xyzzy` everything gets misaligned.
Like you said, if you use a code reformatter it results in spurious and noisy diff.
Do not want.

I am ambivalent about aligning for present/absense of `qualified` (as 
above), but lean against *requiring* it, for similar reasons.

**Comment by romanofski** [comment]
Alright good to know... Then I ditch the formatter :-)

**Comment by frasertweedale** [comment]
👍
